src/App/Fossa/Analyze.hs

```diff
@@ -123,6 +140,60 @@ analyze basedir destination override unpackArchives = runFinally $ do
             ]
           traverse_ (\err -> logError $ "FOSSA error: " <> viaShow err) (uploadError resp)
 
+          contribResult <- Diag.runDiagnostics $ runExecIO $ tryUploadContributors basedir baseurl apiKey $ uploadLocator resp
+          case contribResult of
+            Left failure -> logDebug (Diag.renderFailureBundle failure)
+            Right _ -> pure ()
+
+tryUploadContributors ::
+  ( Has Diag.Diagnostics sig m
+  , Has Exec sig m
+  , MonadIO m
+  ) => Path x Dir
+  -> URI
+  -> ApiKey
+  -> Text -- ^ Locator
+  -> m ()
+tryUploadContributors baseDir baseUrl apiKey locator = do
+  contributors <- fetchGitContributors baseDir
+  uploadContributors baseUrl apiKey locator contributors
+
+
+fetchGitContributors :: 
+  ( Has Diag.Diagnostics sig m
+  , Has Exec sig m
+  , MonadIO m
+  ) => Path x Dir -> m Contributors
+fetchGitContributors basedir = do
+  now <- liftIO getCurrentTime
+  rawContrib <- execThrow basedir $ gitLogCmd now
+  textContrib <- Diag.fromEitherShow . TE.decodeUtf8' $ toStrict rawContrib
+  pure . Contributors . M.map (T.pack . iso8601Show) $ foldr updateMap M.empty $ T.lines textContrib
```

Comment:
This is a really good imperative way of thinking about the problem. I think we can use these functions to replace that `foldr` make it a little more functional (and also eliminate hacks like `minimumDay`):
```hs
M.fromListWith :: Ord k => (a -> a -> a) -> [(k,a)] -> Map k a
-- from Data.Maybe
mapMaybe :: (a -> Maybe b) -> [a] -> [b]
```